Netflix/metaflow-ui
# Run number ws

### Requirements for a pull request


### Description of the Change

Websocket subscriptions for tasks were using the string-format run ID's instead of the run number in the resources field.

e.g. `argo-scheduledsleep-zcc6n` instead of `211533`

The backend does not accept run ID's so the websocket subscription would not provide any messages, leading to the UI not updating.

This only applied to the websocket subscriptions for `/tasks` and `/steps`. All other subscriptions used the run number in their resource path.

The change was to use the `run_number` instead of a run ID for the resource paths for the `/tasks` and `/steps` subscriptions.

### Alternate Designs

Updating the backend to accept run ID.

### Possible Drawbacks

Test for side effects from other parts of the task page that may be affected by these websocket subscriptions. As they were already failing, there shouldn't be anything bad happening.

### Verification Process

* Run a flow
* In MFGUI, check the websocket subscription resources to ensure that the run number, rather than the string run id is present
* Ensure the task status updates

### Release Notes

Change run id to run number for websocket subscriptions for tasks and steps
